lib/tokenizers/tokenizer.ex

```diff
@@ -68,17 +81,51 @@ defmodule Tokenizers.Tokenizer do
       |> Keyword.put(:method, :get)
       |> Keyword.update(:headers, headers, fn existing -> existing ++ headers end)
 
-    case http_client.request(http_opts) do
-      {:ok, response} ->
-        case response.status do
-          status when status in 200..299 ->
-            cache_dir = :filename.basedir(:user_cache, "tokenizers_elixir")
-            :ok = File.mkdir_p(cache_dir)
-            file_path = Path.join(cache_dir, "#{identifier}.json")
+    cache_dir = opts[:cache_dir]
+
+    file_path_fun = fn etag ->
+      Path.join(cache_dir, "#{identifier}-#{opts[:revision]}-#{etag}.json")
```

Comment:
Maybe let's encode the url and etag, just to be sure there are no surprising characters?

```elixir
  defp entry_filename(url, etag) do
    encode_url(url) <> "." <> encode_etag(etag)
  end

  defp encode_url(url) do
    url |> :erlang.md5() |> Base.encode32(case: :lower, padding: false)
  end

  defp encode_etag(etag) do
    Base.encode32(etag, case: :lower, padding: false)
  end
```

Reply:
Yeah, good call!